File: 0435_NonoverlappingIntervals/435.cpp
#include <ostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution
{
public:
    int eraseOverlapIntervals(vector<vector<int>> &intervals)
    {
        if (intervals.empty())
        {
            return 0;
        }

        sort(intervals.begin(), intervals.end(), [](vector<int> &a, vector<int> &b)
             { return a[1] < b[1]; });

        int removed = 0, prev = intervals[0][1];

        intervals.erase(intervals.begin());

        for (auto &i : intervals)
        {
            if (i[0] < prev)
            {
                removed++;
            }
            else
            {
                prev = i[1];
            }
        }
        return removed;
    }
};

File: 0435_NonoverlappingIntervals/435.swift

class Solution {
    func eraseOverlapIntervals(_ intervals: [[Int]]) -> Int {
        return _eraseOverlapIntervals(intervals)
    }
    
    fileprivate func _eraseOverlapIntervals(_ intervals: [[Int]]) -> Int {
        guard !intervals.isEmpty else { return 0 }
        var intervals = intervals.sorted { a, b in
            a[1] < b[1]
        }
        var removed = 0, prev = intervals.removeFirst()[1];
        
        for i in intervals {
            if i[0] < prev {
                removed += 1
            } else {
                prev = i[1]
            }
        }
        return removed
    }
}
